test/controllers/review_controller_test.rb

```diff
@@ -0,0 +1,71 @@
+require "test_helper"
+
+describe ReviewsController do
+  
+  describe "logged in users" do
+    before do
+      perform_login
+    end
+    
+    let (:existing_review) { review(:started) }
+    
+    describe "new" do
+      it "succeeds" do
+        get new_product_review_path(Product.first.id)
+        
+        must_respond_with :success
+      end
+      
+      it "does not allow merchant to review product if product is merchant's" do 
+        product = Product.first
+        merchant = Merchant.find(product.merchant_id)
+        
+        # TODO: Unsure how to modify @current_merchant in reviews_controller.rb for check_authorization() method
```

Comment:
you can pass in a merchant through preform_login and it will set the current merchant!